src/NBitcoin/Block.cs

```diff
@@ -162,21 +167,22 @@ public bool CheckProofOfWork()
 
         public bool CheckProofOfWork(Consensus consensus)
         {
-            consensus = consensus ?? Network.Main.Consensus;
+            NetworkOptions options = consensus?.NetworkOptions ?? NetworkOptions.TemporaryOptions;
```

Comment:
Again lets not allow `consensus.NetworkOptions` to be null, if need be initialise it with temporary options

Reply:
Here we are recognizing the fact that consensus itself may be null as per the legacy behavior of this method. The NetworkOptions already defaults to TemporaryOptions on the Consensus object.